flamedisx/xenon/x1t_sr1.py

```diff
@@ -298,14 +309,27 @@ def random_truth(self, n_events, fix_truth=None, **params):
         d['z_observed'] = d['z']
 
         # Adding some smear according to posrec resolution
-        d['x_observed'] = np.random.normal(d['x_observed'].values, scale=0.4) # 4 mm resolution)
-        d['y_observed'] = np.random.normal(d['y_observed'].values, scale=0.4) # 4 mm resolution)
+        d['x_observed'] = np.random.normal(d['x_observed'].values, scale=4) # 0.4 = 4 mm resolution)
+        d['y_observed'] = np.random.normal(d['y_observed'].values, scale=4) # 4 mm resolution)
+
+        # add effective drift velocity depending on (r,z) position
+        # correct drift time using velocity depending on (r,z) position
+        # z observed is evaluated using default drift velocity as done in data processing
+        if self.variable_drift_velocity:
+            d['drift_velocity'] = self.drift_velocity_map(
+                np.transpose([d['r'].values,
+                              d['z'].values]))
+            d['drift_time'] = -d['z']/d['drift_velocity']
+            d['z_observed'] = -self.default_drift_velocity*d['drift_time'] 
+        else:
+            d['drift_velocity'] = self.default_drift_velocity
+            d['z_observed'] = d['z']
         
         # applying fdc
         delta_r = self.fdc_map(
             np.transpose([d['x_observed'].values,
                           d['y_observed'].values,
-                          d['z_observed'].values,]))
+                          d['z_observed'].values/d['drift_velocity'].values,]))
```

Comment:
Is there an accidental double correction here? I suppose the argument of the FDC map is still (x_observed, y_observed, z_observed), but units of `d['z_observed'].values/d['drift_velocity'].values,]` is time?

Reply:
When we had a call with Francesco Toschi, we found out that the fdc map requires as input two spatial coordinates and the drift time of the event. that is why we introduced the correction. Now it is not clear if by using both drift_velocity and fdc map we are introducing a tautology.. I hope not.. 
this is for sure something to check.
I tag here @eangelino

Reply:
I confirm that this is what we were told by Francesco, which was very non-intuitive. Further, from what I understood, this should be the true drift time (i.e. considering variable drift velocity).